### PR TITLE
Rewrite Ping Passthrough

### PR DESCRIPTION
**Pull request by DoctorMacc**
This commit's primarily changes to using the plugins' APIs for ping information where possible in order to prevent overloading the server with ping requests. This commit also adds:

- A legacy ping passthrough option to handle extreme MOTDs
- A ping passthrough interval option (relevant for legacy passthrough and standalone) to prevent BrokenPipe errors on BungeeCord
- Respond to queries to support services like Mcsrvstatus
- Separate ping passthrough options for MOTD and player counts

Thanks to @theminecoder for implementing proper methods for the plugin versions and @james090500 for implementing the query packet handling.

### CODE REVIEW

**Comment by theminecoder** [comment]
Personally I would just grab the default motd and call the default ping event for each platform rather then trying to do advanced socket things. Means that any plugin based modification can occur and you can merge the 3 seperate calls into 1 standard object for geyser to use. ie for bukkit you can do something like:

```java
ServerListPingEvent event = new ServerListPingEvent(InetAddress.getLocalHost(), Bukkit.getMotd(), Bukkit.getMaxPlayers());
event.callEvent(); // or the normal Bukkit.getPluginManager().callEvent(event);
return new GeyserPingInfo(event.getMotd(), event.getNumPlayers(), event.getMaxPlayers());
````

**Comment by DoctorMacc** [comment]
Yeah, I like that a lot better; I'm just not familiar with any of the plugin APIs. Thanks.

**Comment by LegacyGamerHD** [comment]
how far is this

**Comment by theminecoder** [comment]
@DoctorMacc PR has been made to move the rest of the plugin bootstraps to use internal ping events

**Comment by Hyronymos1337** [comment]
Any eta on this?

**Comment by DoctorMacc** [comment]
Ideally this will be finished tomorrow once I wrap up some loose ends, not including testing.

**Comment by Hyronymos1337** [comment]
I can help you with testing if you send me a build of it

**Comment by DoctorMacc** [comment]
Go to checks up at the top, then click Artifacts near the top-right.

**Comment by muzak23** [comment]
Whenever joining on Bedrock console just has the error 
`>.... [15:57:12 ERROR]: Exception in thread "Thread-2" java.lang.NullPointerException
>.... [15:57:12 ERROR]: at java.base/java.nio.file.Files.provider(Files.java:100)
>.... [15:57:12 ERROR]: at java.base/java.nio.file.Files.newByteChannel(Files.java:370)
>.... [15:57:12 ERROR]: at java.base/java.nio.file.Files.newByteChannel(Files.java:421)
>.... [15:57:12 ERROR]: at java.base/java.nio.file.Files.readAllBytes(Files.java:3205)
>.... [15:57:12 ERROR]: at org.geysermc.floodgate.util.EncryptionUtil.getKeyFromFile(EncryptionUtil.java:67)
>.... [15:57:12 ERROR]: at org.geysermc.connector.network.session.GeyserSession.lambda$authenticate$0(GeyserSession.java:248)
>.... [15:57:12 ERROR]: at java.base/java.lang.Thread.run(Thread.java:834)`
I've tried with remote IP of 0.0.0.0, my docker container, pterodactyl ip, 127.0.0.1, external IP, all have the same outcome. Also, I still get the "IP:#### disconnected" spam when I have legacy ping forwarding on.

**Comment by DoctorMacc** [comment]
^ This user is on Velocity.

**Comment by muzak23** [comment]
Disregard that, I somehow deleted floodgate. That was incredibly stupid haha

File: connector/src/main/java/org/geysermc/connector/network/QueryPacketHandler.java [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+        // Players
+        reply.writeBytes(getPlayers());
+
+        sendPacket(reply);
+    }
+
+    /**
+     * Gets the game data for the query
+     * @return
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/network/QueryPacketHandler.java:133]: Missing return javadoc.

File: connector/src/main/java/org/geysermc/connector/network/QueryPacketHandler.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+        this.token = token;
+    }
+
+    /**
+     * Gets an MD5 token for the current IP/Port.
+     * This should reset every 30 seconds but a new one is generated per instance
+     * Seems wasteful to code something in to clear it when it has no use.
+     * @param token
+     * @param address
+     * @return
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/network/QueryPacketHandler.java:256]: Missing return and params javadoc.

File: connector/src/main/java/org/geysermc/connector/network/QueryPacketHandler.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        this.sessionId = buffer.readInt();
+
+        regenerateToken();
+        handle();
+    }
+
+    /**
+     * Checks the packet is in fact a query packet
+     * @param buffer Query data
+     * @return
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/network/QueryPacketHandler.java:83]: Missing return javadoc.

**Pull request merged**